hedera-node/src/main/java/com/hedera/services/utils/MiscUtils.java

```diff
@@ -411,260 +514,116 @@ public static JKey asFcKeyUnchecked(Key key) {
+	public static Key asKeyUnchecked(final JKey fcKey) {
 		try {
 			return mapJKey(fcKey);
 		} catch (Exception impossible) {
 			return Key.getDefaultInstance();
 		}
 	}
 
-	public static Timestamp asTimestamp(Instant when) {
+	public static Timestamp asTimestamp(final Instant when) {
 		return Timestamp.newBuilder()
 				.setSeconds(when.getEpochSecond())
 				.setNanos(when.getNano())
 				.build();
 	}
 
-	public static Instant timestampToInstant(Timestamp timestamp) {
+	public static Instant timestampToInstant(final Timestamp timestamp) {
 		return Instant.ofEpochSecond(timestamp.getSeconds(), timestamp.getNanos());
 	}
 
-	public static Optional<QueryHeader> activeHeaderFrom(Query query) {
-		switch (query.getQueryCase()) {
-			case TOKENGETNFTINFO:
-				return Optional.of(query.getTokenGetNftInfo().getHeader());
-			case TOKENGETNFTINFOS:
-				return Optional.of(query.getTokenGetNftInfos().getHeader());
-			case TOKENGETACCOUNTNFTINFOS:
-				return Optional.of(query.getTokenGetAccountNftInfos().getHeader());
-			case TOKENGETINFO:
-				return Optional.of(query.getTokenGetInfo().getHeader());
-			case SCHEDULEGETINFO:
-				return Optional.of(query.getScheduleGetInfo().getHeader());
-			case CONSENSUSGETTOPICINFO:
-				return Optional.of(query.getConsensusGetTopicInfo().getHeader());
-			case GETBYSOLIDITYID:
-				return Optional.of(query.getGetBySolidityID().getHeader());
-			case CONTRACTCALLLOCAL:
-				return Optional.of(query.getContractCallLocal().getHeader());
-			case CONTRACTGETINFO:
-				return Optional.of(query.getContractGetInfo().getHeader());
-			case CONTRACTGETBYTECODE:
-				return Optional.of(query.getContractGetBytecode().getHeader());
-			case CONTRACTGETRECORDS:
-				return Optional.of(query.getContractGetRecords().getHeader());
-			case CRYPTOGETACCOUNTBALANCE:
-				return Optional.of(query.getCryptogetAccountBalance().getHeader());
-			case CRYPTOGETACCOUNTRECORDS:
-				return Optional.of(query.getCryptoGetAccountRecords().getHeader());
-			case CRYPTOGETINFO:
-				return Optional.of(query.getCryptoGetInfo().getHeader());
-			case CRYPTOGETLIVEHASH:
-				return Optional.of(query.getCryptoGetLiveHash().getHeader());
-			case CRYPTOGETPROXYSTAKERS:
-				return Optional.of(query.getCryptoGetProxyStakers().getHeader());
-			case FILEGETCONTENTS:
-				return Optional.of(query.getFileGetContents().getHeader());
-			case FILEGETINFO:
-				return Optional.of(query.getFileGetInfo().getHeader());
-			case TRANSACTIONGETRECEIPT:
-				return Optional.of(query.getTransactionGetReceipt().getHeader());
-			case TRANSACTIONGETRECORD:
-				return Optional.of(query.getTransactionGetRecord().getHeader());
-			case TRANSACTIONGETFASTRECORD:
-				return Optional.of(query.getTransactionGetFastRecord().getHeader());
-			case NETWORKGETVERSIONINFO:
-				return Optional.of(query.getNetworkGetVersionInfo().getHeader());
-			default:
-				return Optional.empty();
+	public static Optional<QueryHeader> activeHeaderFrom(final Query query) {
```

Comment:
The overhead of this implementation will be **staggeringly** high compared to the current code (both in terms of object creations and CPU cycles). 😞 

I definitely get that the `if...else if...else if...` pattern isn't the prettiest, but since this method is called a ton, I'd suggest we stay with lean-and-mean here.

Reply:
Right here is a runtime search and found method, but could also be changed into static initialization.